packages/openapi-code-generator/src/typescript/common/type-builder.ts

```diff
@@ -33,6 +34,61 @@ export type TypeBuilderConfig = {
   allowAny: boolean
 }
 
+type IRTypeIntersection = {type: "type-intersection"; types: IRType[]}
+type IRTypeUnion = {type: "type-union"; types: IRType[]}
+type IRTypeOther = string
```

Comment:
Eventually will probably aim to leave the typescript types (and schema code) in a "non-stringy" representation all the way through to the emit phase.

This will be particularly important for contextual alterations (eg; readonly / writeonly modifiers)